twilio/request_validator.py

```diff
@@ -82,17 +96,29 @@ def validate(self, uri, params, signature):
             params = {}
 
         parsed_uri = urlparse(uri)
+        uri_with_port = uri
+        uri_without_port = uri
+
         if parsed_uri.scheme == "https" and parsed_uri.port:
-            uri = remove_port(parsed_uri)
+            uri_without_port = remove_port(parsed_uri)
+        elif parsed_uri.scheme == "https":
+            uri_with_port = add_port(parsed_uri, 443)
+        elif parsed_uri.scheme == "http" and parsed_uri.port:
+            uri_without_port = remove_port(parsed_uri)
+        elif parsed_uri.scheme == "http":
+            uri_with_port = add_port(parsed_uri, 80)
```

Comment:
Should target something like this:
```
uri_with_port = add_port(uri)
uri_without_port = remove_port(uri)
```
I think `remove_port` works fine as-is. `add_port` will need to add the appropriate port if it's missing.

Reply:
Making sure I understand this comment: move the logic of adding/removing ports to the appropriate add/remove functions?

Reply:
Correct. They're independent so it makes sense to isolate the logic. And it makes adding tests for each simple.